### PR TITLE
Missing Node test cases

### PR DESCRIPTION
**Pull request by eswarasai**
### Description
Added test cases for missing node

Fixes: #835

### CODE REVIEW

File: testworld/missing_node_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+func TestMissingNode_InvalidIdentity(t *testing.T) {
+	t.Parallel()
+
+	// Hosts
+	alice := doctorFord.getHostTestSuite(t, "Alice")
+
+	// RandomDID
+	randomDID := testingidentity.GenerateRandomDID()
+
+	// alice shares a document with bob and charlie
```
Comment (vimukthi-git) [inline testworld/missing_node_test.go]: fix comment

**Comment by vimukthi-git** [comment]
I just noticed that the second test here returns at point from the node earlier than expected, so the document is not getting anchored it seems. We can still merge the pr if this is the current buggy behaviour of the node. @eswarasai could you investigate why the node returns an error before anchoring the document?

File: testworld/missing_node_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+	// Alice shares a document with a randomly generated DID with missing P2P Key
+	res := createDocument(alice.httpExpect, alice.id.String(), typeInvoice, http.StatusOK, defaultInvoicePayload([]string{randomDID.String()}))
+	txID := getTransactionID(t, res)
+	status, message := getTransactionStatusAndMessage(alice.httpExpect, alice.id.String(), txID)
+	if status != "failed" {
+		t.Error(message)
+	}
+
+	assert.Contains(t, message, "missing p2p key")
```
Comment (vimukthi-git) [inline testworld/missing_node_test.go]: As explained on DM lets add the correct error string here and also a check for the anchor on chain. 
Thank you 👍

**Comment by eswarasai** [comment]
@vimukthi-git @vedhavyas -- This PR should be good to go now. I've rebased with `develop` again. In case the tests fail, please re-run them by clearing cache in Travis CI. Let me know if there's anything else pending. Thanks :)

cc: @lucasvo

File: identity/ideth/service.go [hunk 3]
```diff
@@ -292,6 +292,10 @@ func (i service) CurrentP2PKey(did id.DID) (ret string, err error) {
 		return ret, err
 	}
 
+	if len(keys) == 0 {
+		return "", errors.New("missing p2p key")
```
Comment (vimukthi-git) [inline identity/ideth/service.go]: 👍

**Pull request merged**